common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java

```diff
@@ -270,6 +273,34 @@ public void loadAvatar(@LuaNotNil String playerUUID) {
         NetworkStuff.getUser(new UserData(uuid));
     }
 
+    @LuaWhitelist
+    @LuaMethodDoc(
+      overloads = {
+        @LuaMethodOverload(
+          argumentTypes = { String.class },
+          argumentNames = { "avatarUUID" }
+        )
+      },
+      value = "goofy.get_avatar_nameplate"
+    )
+    public String[] getAvatarNameplate(@LuaNotNil String avatarUUID) {
+      UUID uuid = UUID.fromString(avatarUUID);
+      Avatar avatar = AvatarManager.getLoadedAvatar(uuid);
+
+      if (avatar == null) {
+        return new String[]{avatarUUID, avatarUUID, avatarUUID};
+      }
+
+      NameplateAPI plate = avatar.luaRuntime.nameplate;
+
+      String name = avatar.entityName;
+      String chat = ObjectUtils.firstNonNull(plate.CHAT.getText(), name, avatarUUID);
+      String entity = ObjectUtils.firstNonNull(plate.ENTITY.getText(), name, avatarUUID);
+      String list = ObjectUtils.firstNonNull(plate.LIST.getText(), name, avatarUUID);
+
+      return new String[]{chat, entity, list};
```

Comment:
personally I think this should be a named table

Reply:
that is how you return multiple things on the lua side

Reply:
and you're only telling me this now?? i thought you needed to construct a `Varargs`